src/Mutant/Generator/MutationsGenerator.php

```diff
@@ -146,7 +152,11 @@ private function getMutationsFromFile(SplFileInfo $file, bool $onlyCovered, arra
         $traverser->addVisitor(new FullyQualifiedClassNameVisitor());
         $traverser->addVisitor(new ReflectionVisitor());
 
-        $traverser->traverse($initialStatements);
+        try {
+            $traverser->traverse($initialStatements);
+        } catch (MutatorException $e) {
```

Comment:
catching a mutator exception to thrown another mutator exception is a bit of a smell to me. Since it's a hard error (i.e. we are stopping the process on that error), I think an `InvalidArgumentException` exception is valid, we just need a comprehensive error message

Reply:
I agree it is a bit of a smell, but i'm not sure what other option would be the best.

I don't agree with changing it to an `InvalidArgumentException`, as this could cause confusion to our end user. I'd rather name it something like `InternalInfectionException`

Reply:
IMO just throwing a meaningful exception is enough: we don't need two of them especially since you don't really enrich the message. So either would throw the child one and remove this catch block remove the child one and throw the exception here.

>I don't agree with changing it to an InvalidArgumentException, as this could cause confusion to our end user. I'd rather name it something like InternalInfectionException

I don't know what's confusing: `InvalidArgumentException` is a `LogicException`  which means there is something that requires a fix and not something that should be catch. Having a dedicated exception here brings no benefit and don't make it clearer if the exception message is clear enough